addons/osfstorage/tests/test_views.py

```diff
@@ -649,6 +665,30 @@ def test_move_checkedout_file(self):
         )
         assert_equal(res.status_code, 405)
 
+    def test_within_node_move_while_preprint(self):
```

Comment:
These three tests were what I had included originally. There are some more edge cases that would ideally be covered. A followup ticket for that (and the note about is_preprint_primary) is probably needed.